src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,171 @@
+            currentValue.next.prev = newNode;
+            newNode.next = currentValue.next;
+            head = newNode;
+        } else if (index == size) {
+            currentValue.prev.next = newNode;
+            newNode.prev = currentValue.prev;
+            tail = newNode;
+        } else {
+            newNode.prev = currentValue.prev;
+            newNode.next = currentValue.next;
+            currentValue.prev.next = newNode;
+            currentValue.next.prev = newNode;
+        }
+        currentValue.prev = null;
+        currentValue.next = null;
+        return currentValue.item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> currentNode = getNodeByIndex(index);
+        unlink(currentNode);
+        return currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        return removeNodeByValue(object);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void validateIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Invalid index " + index);
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        validateIndex(index);
+        int count = 0;
+        Node<T> currentNode = head;
+        while (count++ != index) {
+            currentNode = currentNode.next;
+        }
+        return currentNode;
+    }
+
+    private boolean removeNodeByValue(T value) {
+        Node<T> currentNode = head;
+        boolean result = false;
+        int count = 0;
+        while (count++ != size) {
+            if (currentNode.item != null) {
+                if (!currentNode.item.equals(value)) {
+                    currentNode = currentNode.next;
+                } else {
+                    return unlink(currentNode);
+                }
+            } else if (value == null) {
+                return unlink(currentNode);
+            } else {
+                currentNode = currentNode.next;
+            }
+        }
+        return result;
+    }
+
+    private boolean unlink(Node<T> node) {
+        if (node != null) {
+            if (node.prev == null && node.next == null) {
+                head = null;
+                tail = null;
+            } else if (node.prev == null) {
+                node.next.prev = null;
+                head = node.next;
+            } else if (node.next == null) {
+                node.prev.next = null;
+                tail = node.prev;
+            } else {
+                Node<T> previous = node.prev;
+                Node<T> next = node.next;
+                previous.next = next;
+                next.prev = previous;
+                node.prev = null;
+                node.next = null;
+            }
+            size--;
+        }
+        return true;
```

Comment:
invert condition to avoid big code block identing
```suggestion
        if (node == null) {
        return true;
        }
            if (node.prev == null && node.next == null) {
                head = null;
                tail = null;
            } else if (node.prev == null) {
                node.next.prev = null;
                head = node.next;
            } else if (node.next == null) {
                node.prev.next = null;
                tail = node.prev;
            } else {
                Node<T> previous = node.prev;
                Node<T> next = node.next;
                previous.next = next;
                next.prev = previous;
                node.prev = null;
                node.next = null;
            }
            size--;
```